src/service-api.ts

```diff
@@ -0,0 +1,326 @@
+        throw new InvalidH5PFileError(h5pFile.mimetype);
+      }
+      */
+
+      const contentId = v4();
+      const targetFolder = path.join(__dirname, TMP_EXTRACT_DIR, contentId);
+      const remoteRootPath = buildRootPath(contentId);
+
+      await mkdir(targetFolder, { recursive: true });
+
+      // try-catch block for local storage cleanup
+      try {
+        const savePath = buildH5PPath(targetFolder, contentId);
+        const contentFolder = buildContentPath(targetFolder);
+
+        // save H5P file
+        await pipeline(h5pFile.file, fs.createWriteStream(savePath));
+        await extract(savePath, { dir: contentFolder });
+
+        const result = await h5pValidator.validatePackage(contentFolder);
+        if (!result.isValid) {
+          throw new InvalidH5PFileError(result.error);
+        }
+
+        // try-catch block for remote storage cleanup
+        try {
+          // upload whole folder to public storage
+          await uploadH5P(targetFolder, remoteRootPath, member);
+
+          const item = await createH5PItem(
+            h5pFile.filename,
+            contentId,
+            remoteRootPath,
+            member,
+            parentId,
+          );
+          return item;
+        } catch (error) {
+          // delete public storage folder of this H5P if upload or creation fails
+          fileTaskManager.createDeleteFolderTask(member, {
+            folderPath: remoteRootPath,
+          });
+
+          // remove local temp folder, before rethrowing
+          rm(targetFolder, { recursive: true });
+
+          // log and rethrow to let fastify handle the error response
+          log.error('graasp-plugin-h5p: unexpected error occured while importing H5P:');
+          log.error(error);
+          throw error;
+        }
+        // end of try-catch block for remote storage cleanup
+      } finally {
+        // in all cases, remove local temp folder
+        rm(targetFolder, { recursive: true });
+      }
+      // end of try-catch block for local storage cleanup
+    },
+  );
+
+  /**
+   * Delete H5P assets on item delete
+   */
+  const deleteItemTaskName = itemTaskManager.getDeleteTaskName();
+  taskRunner.setTaskPostHookHandler<Item<H5PExtra>>(deleteItemTaskName, async (item, actor) => {
+    if (item.type !== H5P_ITEM_TYPE) {
+      return;
+    }
+    const deleteTask = fileTaskManager.createDeleteFolderTask(actor, {
+      folderPath: buildRootPath(item.extra.h5p.contentId),
+    });
+    await taskRunner.runSingle(deleteTask);
+  });
+
+  /**
+   * Copy H5P assets on item copy
+   */
+  const copyItemTaskName = itemTaskManager.getCopyTaskName();
+  taskRunner.setTaskPreHookHandler<Item<H5PExtra>>(copyItemTaskName, async (item, actor) => {
+    if (item.type !== H5P_ITEM_TYPE) {
+      return;
+    }
+    if (!item.extra?.h5p) {
+      throw new Error('Invalid state: missing previous H5P item extra on copy');
+    }
+
+    const contentId = v4();
```

Comment:
Here is it supposed to be a random id or the new item id?

Reply:
Content ID is uniquely generated when the h5p files are uploaded or copied. It is independent from item IDs.